debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/NonStreamingWal2JsonMessageDecoder.java

```diff
@@ -51,7 +51,7 @@ public void processMessage(ByteBuffer buffer, ReplicationMessageProcessor proces
             final byte[] content = Arrays.copyOfRange(source, buffer.arrayOffset(), source.length);
             final Document message = DocumentReader.floatNumbersAsTextReader().read(content);
             LOGGER.debug("Message arrived for decoding {}", message);
-            final int txId = message.getInteger("xid");
+            final long txId = message.getLong("xid");
```

Comment:
Can you do the equivalent adjustments to `StreamingWal2JsonMessageDecoder`, too?

Reply:
yes of course